src/app.rs

```diff
@@ -707,6 +707,35 @@ impl App {
         }
     }
 
+    pub fn toggle_save_track(&mut self, track_id: String) {
+        if let Some(spotify) = &self.spotify {
+            match spotify.current_user_saved_tracks_contains(&[track_id.clone()]) {
+                Ok(saved) => {
+                    if saved[0] {
```

Comment:
This is probably overkill, but to be safe let's use `saved.first()` instead of the array accessor `[0]`. We'd then need to handle the option.